runtime/src/resource_tracker.rs

```diff
@@ -63,67 +63,54 @@ macro_rules! log {
 /// Helper function to get a resource from a builder.
 ///
 /// This function is called by the loader (see codegen) to create each type of needed resource.
-pub async fn get_resource<B, T, O>(
+pub async fn get_resource<B, O, R>(
     builder: B,
     factory: &mut ProvisionerFactory,
     resource_tracker: &mut ResourceTracker,
-) -> Result<T, shuttle_service::Error>
+) -> Result<O, shuttle_service::Error>
 where
-    B: ResourceBuilder<T, Output = O>,
-    O: Serialize + DeserializeOwned,
+    B: ResourceBuilder<Output = O>,
+    O: Serialize + DeserializeOwned + IntoResource<R>,
```

Comment:
I'm fine with this generic, just want to double check if we can potentially remove it?

Reply:
I don't see a way of doing it. Open to suggestions. `R` is the final output type getting into the user's main function, so it feels like it must be generic.

Reply:
You mean the `init()` call won't work in the user's main function right?